app/views/events/_apply_button.erb

```diff
@@ -0,0 +1,7 @@
+<% if current_user.present? && current_user.application_letter_for_event?(event) %>
+    <%= link_to t("helpers.links.show_application"), check_application_letter_path(current_user.application_letter_for_event(event)),
+                :class => 'btn btn-default btn-sm' %>
+<% elsif not event.after_deadline? %>
```

Comment:
This `current_user.application_letter_for_event?(event)` checks if for the current user exists an application letter for the event. If no application letter exists the button will not be displayed.